### ISSUE TITLE
Inconsistent import/prefer-default-export with Flow type

### ISSUE THREAD

**Issue by rodrigobdz** [benmosher/eslint-plugin-import#1164]
The following file is generating a warning that default export should be used. 
> Prefer default export. (import/prefer-default-export)

```js
/* @flow */

export type AnswerState = {
  foo: string
}

export const answer: AnswerState = {
  foo: ''
}
```

This one however, is not generating any errors or warnings.

```diff
/* @flow */

- export type AnswerState = {
+ type AnswerState = {
  foo: string
}

export const answer: AnswerState = {
  foo: ''
}

+ export type { AnswerState }
```

Originally posted in [#484](https://github.com/benmosher/eslint-plugin-import/issues/484#issuecomment-415357490).
